### PR TITLE
Chronos : Detect low-quality data and provide suggestions

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

We have met with a large amount of different time series, and we should detect the low-quality data and provide suggestion (e.g. call .impute or .resample) during the tsdataset initialization

### 1. Why the change?

https://github.com/intel-analytics/BigDL/issues/5843, reletaed discussion https://github.com/intel-analytics/BigDL/issues/6023

### 2. User API changes

No change now.

### 3. Summary of the change 

- [x]  Have irregular interval among the data
 - [x] datetime col is not transformed to timestamp type
 - [x] Missing value > 20%
 - [ ] One/multiple target does not have any pattern (very-low seasonality or too much noice)

### 4. How to test?
- [ ] Unit test
- [ ] Application test

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    '''
+    detect the low-quality data and provide suggestion (e.g. call .impute or .resample).
+
+    :param df: a pandas dataframe for your raw time series data.
+    :param dt_col: a str indicates the col name of datetime
+            column in the input data frame, the dt_col must be sorted
+            from past to latest respectively for each id.
+    '''
+    invalidInputError(dt_col in df.columns, f"dt_col {dt_col} can not be found in df.")
+    invalidInputError(pd.isna(df[dt_col]).sum() == 0, "There is N/A in datetime col")
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:33]: https://github.com/rnwang04/BigDL/blob/low_quality_detection/python/chronos/src/bigdl/chronos/data/tsdataset.py#L964
https://github.com/rnwang04/BigDL/blob/low_quality_detection/python/chronos/src/bigdl/chronos/data/tsdataset.py#L979
These 2 have been checked.

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    # 3. missing value check
+    flag = flag and _missing_value_check(df, dt_col)
+
+    # 4. pattern check and noise check
+    # TODO:
+
+    return flag
+
+
+def _timestamp_type_check(df_column):
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:50]: add comment here to explain what this checker is used for

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        return False
+    return True
+
+
+def _time_interval_check(df_column):
+    interval = df_column.shift(-1) - df_column
+    intervals = interval[:-1].unique()
+    if len(intervals) > 1:
+        logging.warning("There are irregular interval(more than one interval length)"
+                        " among the data")
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:63]: please call `.resample(interval).impute()` first to clean the data.

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py (hunk 1 above)
Comment (rnwang04) [reply python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:33]: will remove them.

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py (hunk 3 above)
Comment (rnwang04) [reply python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py]: done.

File: python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py (hunk 2 above)
Comment (rnwang04) [reply python/chronos/src/bigdl/chronos/data/utils/quality_inspection.py:48]: done.

**Pull request merged**